Result/Result.swift

```diff
@@ -179,6 +179,15 @@ public func ?? <T, Error> (left: Result<T, Error>, @autoclosure right: () -> Res
 	return left.recoverWith(right())
 }
 
+// MARK: - Derive result from failable closure
+
+public func materialize<T, E: ErrorType>(f: () throws -> T) -> Result<T, E> {
+    do {
+        return .success(try f())
+    } catch {
+        return .failure(error as! E)
```

Comment:
Yea, the force cast is required. The `as NSError` works because `ErrorType` actually requires `_domain` and `_code` properties that are used to fill out the `NSError`. In the case of enums, the compiler automatically generates values for said properties.